### PR TITLE
Adding scalar to the c10 registration type check

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32886 Adding scalar to the c10 registration type check**

Differential Revision: [D19673484](https://our.internmc.facebook.com/intern/diff/D19673484)

### CODE REVIEW

File: torch/csrc/jit/register_c10_ops.cpp [hunk 1]
```diff
@@ -63,6 +63,16 @@ Operator createOperatorFromC10(const c10::OperatorHandle& op) {
...
+          } else if (type->kind() == TypeKind::NumberType) {
+            AT_ASSERT(iter->isDouble() || iter->isInt());
+            if (iter->isDouble()) {
+              tracer::addInputs(node, args[i].name().c_str(), iter->toDouble());
+            } else if (iter->isInt()) {
+              tracer::addInputs(node, args[i].name().c_str(), iter->toInt());
+            } else {
+              throw std::runtime_error(
+                "unsupported input list type: " + elem_type->str());
+            }
```
Comment (jamesr66a) [inline torch/csrc/jit/register_c10_ops.cpp]:
```
          } else if (type->kind() == TypeKind::NumberType) {
            tracer::addInputs(node, args[i].name().c_str(), iter->toScalar());
          } ...
```

The type checking and unwrapping will be done internally

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 6399c0f6:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=6399c0f6dc7450cf1b5e538b0b26a2a60dc193c0).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4393530) pytorch_linux_xenial_py3_clang5_mobile_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=93927487) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=93927487))
<details>
<summary>
<code>Jan 31 23:21:03 /var/lib/jenkins/workspace/torch/csrc/jit/register_c10_ops.cpp:74:51: error: 'elem_type' was not declared in this scope</code>
</summary>

```
Jan 31 23:21:00 [ 86%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inliner.cpp.o 
Jan 31 23:21:01 [ 86%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/lift_closures.cpp.o 
Jan 31 23:21:01 [ 86%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inline_forked_closures.cpp.o 
Jan 31 23:21:01 [ 86%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/dead_code_elimination.cpp.o 
Jan 31 23:21:01 [ 87%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/decompose_ops.cpp.o 
Jan 31 23:21:02 [ 87%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/canonicalize_ops.cpp.o 
Jan 31 23:21:02 [ 87%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/erase_number_types.cpp.o 
Jan 31 23:21:02 [ 87%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/fixup_trace_scope_blocks.cpp.o 
Jan 31 23:21:02 [ 87%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inline_fork_wait.cpp.o 
Jan 31 23:21:03 /var/lib/jenkins/workspace/torch/csrc/jit/register_c10_ops.cpp: In lambda function: 
Jan 31 23:21:03 /var/lib/jenkins/workspace/torch/csrc/jit/register_c10_ops.cpp:74:51: error: 'elem_type' was not declared in this scope 
Jan 31 23:21:03                  "unsupported input list type: " + elem_type->str()); 
Jan 31 23:21:03                                                    ^ 
Jan 31 23:21:03 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/register_c10_ops.cpp.o] Error 1 
Jan 31 23:21:03 make[2]: *** Waiting for unfinished jobs.... 
Jan 31 23:21:03 caffe2/CMakeFiles/torch_cpu.dir/build.make:7561: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/register_c10_ops.cpp.o' failed 
Jan 31 23:21:03 CMakeFiles/Makefile2:903: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed 
Jan 31 23:21:03 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2 
Jan 31 23:21:03 Makefile:138: recipe for target 'all' failed 
Jan 31 23:21:03 make: *** [all] Error 2 
Jan 31 23:21:03 + sccache_epilogue 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Review by jamesr66a — APPROVED** [review]
Looks fine but better with a test

**Comment by z-a-f** [comment]
> Looks fine but better with a test

Adding an issue, will implement when I am back

**Pull request closed without merging**